### PR TITLE
Update readme file

### PR DESCRIPTION
**Pull request by wattli**
**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by myidpt** [comment]
I think our integration test fails. I'm trying to fix it.

File: README.md [hunk 1]
```diff
@@ -3,161 +3,163 @@
...
+1.  Istio CA watches Kubernetes API Server, creates a [SPIFFE](https://spiffe.github.io/docs/svid) key and certificate pair for each of the existing and new service accounts, and sends them to API Server.
+
+1.  When a pod is created, API Server mounts the key and certificate pair according to the service account using [Kubernetes secrets](https://kubernetes.io/docs/concepts/configuration/secret/).
+
+1.  [Pilot]({{home}}/docs/concepts/traffic-management/pilot.html) generates the config with proper key and certificate and secure naming information,
+which
+ defines what service account(s) can run a certain service, and passes it to Envoy.
+
+### Deployment phase (VM/bare-metal Machines Scenario)
+
```
Comment (myidpt) [inline README.md]: We may need to add the secure naming configuration for this special use case, i.e., adding the service account as the service annotation.

File: README.md [hunk 2]
```diff
@@ -3,161 +3,163 @@
...
-  “team1-ns” namespace for team1, and “team2-ns” namespace for team2, such that
-  both teams won’t be able to access each other’s services.
-- If Istio CA is compromised, all its managed key & cert in the cluster may be
-  exposed. We strongly recommend to run Istio CA on a dedicated namespace
-  (e.g., istio-ca-ns) which only cluster admins have access to.
-  - We are looking into running Isito CA in the kubernetes master in the future releases.
+The Istio Auth workflow consists of two phases, deployment and runtime. For the deployment phase, we discuss the two scenarios (i.e., in Kubernetes and VM/bare-metal machines) separately since they are different. Once the key and certificate are deployed, the runtime phase is the same for the two scenarios. We briefly cover the workflow in this section.
+
+### Deployment phase (Kubernetes Scenario)
+
```
Comment (myidpt) [inline README.md]: Do we need multiple empty lines (also below)?

File: README.md [hunk 3]
```diff
@@ -3,161 +3,163 @@
...
-- Secure Istio components (Mixer, Istio-Manager, etc.)
-- Inter-cluster service-to-service authentication
-- End-user to service authentication using JWT/OAuth2/OpenID_Connect
-- Support GCP service account and AWS service account
-- Non-http traffic (MySql, Redis, etc.) support
-- Auth info propagation from Envoy to the service
-- Unix domain socket for local communication between service and Envoy
-- Middle proxy support
-- Pluggable key management component
+Let's consider a 3-tier application with three services: photo-frontend, photo-backend, and datastore. Photo-frontend and photo-backend services are managed by the photo SRE team while the datastore service is managed by the datastore SRE team. Photo-frontend can access photo-backend, and photo-backend can access datastore. However, photo-frontend cannot access datastore.
```
Comment (myidpt) [inline README.md]: What is a 3-tier application?

File: README.md [hunk 4]
```diff
@@ -3,161 +3,163 @@
...
+
+*   Secure Istio components (Mixer, Pilot)
+
+*   End-user to service authentication using JWT/OAuth2/OpenID_Connect.
+
+*   Support GCP service account
+
+*   Unix domain socket for local communication between service and Envoy
+
+*   Middle proxy support
```
Comment (myidpt) [inline README.md]: Layer-7 middle proxy support

File: README.md [hunk 5]
```diff
@@ -3,161 +3,163 @@
...
+
+
+
+*   Providing each service with a strong identity that represents its role to enable interoperability across clusters and clouds
+
+*   Securing service to service communication and end-user to service communication
+
+*   Providing a key management system to automate key and certificate generation, distribution, rotation, and revocation
+
+*   Upcoming features:
```
Comment (myidpt) [inline README.md]: I think these are covered by "Future Work".

File: README.md [hunk 6]
```diff
@@ -3,161 +3,163 @@
...
+datastore SRE team creates 1 service account to run the datastore service in namespace datastore-ns. Moreover, we need to enforce the service access control
+in [Istio Mixer]({{home}}/docs/concepts/policy-and-control/mixer.html) such that photo-frontend cannot access datastore.
+
+In this setup, Istio CA is able to provide keys and certificates management for all namespaces, and isolate microservice deployments from each other.
+
+## Future work
+
+*   Inter-cluster service-to-service authentication
+
+*   Powerful authorization mechanisms: ABAC, RBAC, etc
```
Comment (myidpt) [inline README.md]: Hmm... RBAC and ABAC seems like general authz approaches, are they more powerful than others?

File: README.md (hunk 5 above)
Comment (wattli) [reply README.md]: Good point, removed.

File: README.md (hunk 2 above)
Comment (wattli) [reply README.md]: Removed.

File: README.md (hunk 1 above)
Comment (wattli) [reply README.md]: good point, would you like to make a follow-up PR :)

Comment (wattli) [reply README.md]: Good point, done.

File: README.md (hunk 3 above)
Comment (wattli) [reply README.md]: ? It means we have frontend, backend, datastore.

Comment (wattli) [reply README.md]: It's not a new change, btw.

File: README.md (hunk 6 above)
Comment (wattli) [reply README.md]: I think we want to deliver the impression to readers. It's actually added by wencheng :)

File: README.md (hunk 4 above)
Comment (wattli) [reply README.md]: Done.

File: README.md (hunk 6 above)
Comment (myidpt) [reply README.md]: I see :)

**Pull request merged**